eks-anywhere-common/Testers/Pulumi/pulumi-tester-job.yaml

```diff
@@ -0,0 +1,169 @@
+
+    echo ""
+    echo "Writing out test manifest file '${MANIFEST_FILENAME}'"
+    # Note that while we use a random Pulumi stack name within the Stack resource,
+    # we keep the name of the Kubernetes Stack and Program resources static. This is
+    # intentional. If a test fails, we don't want it to leave superfluous Stack and
+    # Program K8s resources behind because the operator will keep trying to
+    # reconcile them. (The test failed - they should never be re-run again.)
+    # Instead, we want to reuse the same Kubernetes resource over and over but have
+    # it generate a new, uniquely-named Pulumi stack.
+    cat << EOF > $MANIFEST_FILENAME
+    apiVersion: pulumi.com/v1
+    kind: Program
+    metadata:
+      name: eks-pulumi-operator-test
+      namespace: pulumi
+    program:
+      resources:
+        myRandomPet:
+          type: random:RandomPet
+      outputs:
+        petName: \${myRandomPet.id}
+    ---
+    apiVersion: pulumi.com/v1
+    kind: Stack
+    metadata:
+      name: eks-pulumi-operator-test
+      namespace: pulumi
+    spec:
+      stack: ${STACKPATH}
+      programRef:
+        name: eks-pulumi-operator-test
+      destroyOnFinalize: true
+      envRefs:
+        PULUMI_ACCESS_TOKEN:
+          type: Secret
+          secret:
+            name: pulumi-access-token
+            key: value
+    EOF
+
+
+    echo ""
+    echo "Deploying sample stack and program."
+    kubectl apply -f $MANIFEST_FILENAME
+
+    echo ""
+    echo "Waiting for the operator to deploy the stack."
+    sleep 10
+
+    echo ""
+    echo "Verifying that the stack exists."
+    curl \
+      --fail \
+      -H "Accept: application/vnd.pulumi+8" \
+      -H "Content-Type: application/json" \
+      -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+      https://api.pulumi.com/api/stacks/${STACKPATH}
+
+    echo ""
+    echo "Destroying K8s Stack resource"
+    kubectl delete -n pulumi stacks/eks-pulumi-operator-test
+
+    echo ""
+    echo "Waiting for the operator to remove the stack"
+    sleep 10
+
+    echo ""
+    echo "Verifying the stack no longer exists"
+    STATUSCODE=$(curl \
+      -s \
+      -o /dev/null \
+      --w "%{http_code}" \
+      -H "Accept: application/vnd.pulumi+8" \
+      -H "Content-Type: application/json" \
+      -H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+      https://api.pulumi.com/api/stacks/${STACKPATH}
+    )
+
+    if test $STATUSCODE -ne 404; then
+      echo "ERROR: Expected HTTP status code 404 from the Pulumi Cloud API when querying the stack. Got HTTP status code $STATUSCODE instead."
+      false
+    fi
+
+    # This is for purely for running the script locally. Since the K8s tester Job is
+    # run in an ephemeral container, deleting the file is unnecessary in that
+    # context:
+    echo ""
+    echo "Deleting test manifest file '${MANIFEST_FILENAME}'"
+    rm ${MANIFEST_FILENAME}
+---
+apiVersion: batch/v1
+kind: Job
```

Comment:
This should be a CronJob with a Schdule. Please check our partner submission for sample.

Reply:
Fixed.